mm2src/coins/utxo/qrc20.rs

```diff
@@ -0,0 +1,698 @@
+    }
+
+    fn wait_for_tx_spend(&self, transaction: &[u8], wait_until: u64, from_block: u64) -> TransactionFut {
+        utxo_common::wait_for_tx_spend(&self.utxo_arc, transaction, wait_until, from_block)
+    }
+
+    fn tx_enum_from_bytes(&self, bytes: &[u8]) -> Result<TransactionEnum, String> {
+        utxo_common::tx_enum_from_bytes(bytes)
+    }
+
+    fn current_block(&self) -> Box<dyn Future<Item=u64, Error=String> + Send> {
+        utxo_common::current_block(&self.utxo_arc)
+    }
+
+    fn address_from_pubkey_str(&self, pubkey: &str) -> Result<String, String> {
+        utxo_common::address_from_pubkey_str(self, pubkey)
+    }
+
+    fn display_priv_key(&self) -> String {
+        utxo_common::display_priv_key(&self.utxo_arc)
+    }
+}
+
+impl MmCoin for Qrc20Coin {
+    fn is_asset_chain(&self) -> bool {
+        utxo_common::is_asset_chain(&self.utxo_arc)
+    }
+
+    fn can_i_spend_other_payment(&self) -> Box<dyn Future<Item=(), Error=String> + Send> {
+        utxo_common::can_i_spend_other_payment()
+    }
+
+    fn withdraw(&self, req: WithdrawRequest) -> Box<dyn Future<Item=TransactionDetails, Error=String> + Send> {
+        Box::new(qrc20_withdraw(self.clone(), req).boxed().compat())
+    }
+
+    fn decimals(&self) -> u8 {
+        utxo_common::decimals(&self.utxo_arc)
+    }
+
+    fn process_history_loop(&self, ctx: MmArc) {
+        utxo_common::process_history_loop(self, ctx)
+    }
+
+    fn tx_details_by_hash(&self, hash: &[u8]) -> Box<dyn Future<Item=TransactionDetails, Error=String> + Send> {
+        utxo_common::tx_details_by_hash(self.clone(), hash)
+    }
+
+    fn history_sync_status(&self) -> HistorySyncState {
+        utxo_common::history_sync_status(&self.utxo_arc)
+    }
+
+    fn get_trade_fee(&self) -> Box<dyn Future<Item=TradeFee, Error=String> + Send> {
+        utxo_common::get_trade_fee(self.clone())
+    }
+
+    fn required_confirmations(&self) -> u64 {
+        utxo_common::required_confirmations(&self.utxo_arc)
+    }
+
+    fn requires_notarization(&self) -> bool {
+        utxo_common::requires_notarization(&self.utxo_arc)
+    }
+
+    fn set_required_confirmations(&self, confirmations: u64) {
+        utxo_common::set_required_confirmations(&self.utxo_arc, confirmations)
+    }
+
+    fn set_requires_notarization(&self, requires_nota: bool) {
+        utxo_common::set_requires_notarization(&self.utxo_arc, requires_nota)
+    }
+
+    fn my_unspendable_balance(&self) -> Box<dyn Future<Item=BigDecimal, Error=String> + Send> {
+        // QRC20 cannot have unspendable balance
+        Box::new(futures01::future::ok(0.into()))
+    }
+}
+
+async fn qrc20_withdraw(coin: Qrc20Coin, req: WithdrawRequest) -> Result<TransactionDetails, String> {
+    let to_addr = match &coin.utxo_arc.address_format {
+        UtxoAddressFormat::Standard => try_s!(Address::from_str(&req.to)),
+        UtxoAddressFormat::CashAddress { .. } => try_s!(Address::from_cashaddress(
```

Comment:
QTUM/QRC20 do not support cash addresses for sure, we can simply use `Address::from_str` here and also maybe deny QRC20 coin initialization if it has cash address format in config for some reason.

Reply:
Remove the cashaddress for QRC20